### PR TITLE
feat: search transactions by learner email and content title

### PR DESCRIPTION
**Pull request by johnnagro**
### Description

- DRF search on Transactions list
- search fields include learner email address and content title
- https://2u-internal.atlassian.net/browse/ENT-7683

### CODE REVIEW

File: enterprise_subsidy/apps/api/v1/tests/test_views.py [hunk 1]
```diff
@@ -1353,6 +1354,56 @@ def test_fetch_transaction_with_external_reference(self):
...
+    def test_list_search(self, role, request_query_params, expected_response_status, expected_response_uuids):
+        """
+        Test list Transactions permissions.
+        """
+        if role == "admin":
+            self.set_up_admin()
+        elif role == "learner":
+            self.set_up_learner()
+        elif role == "operator":
+            self.set_up_operator()
```
Comment (pwnage101) [inline enterprise_subsidy/apps/api/v1/tests/test_views.py:1392]: Nit: learner and operator not needed here or in the ddt.data()

**Review by pwnage101 — APPROVED** [review]

**Comment by pwnage101** [comment]
(fix failing test first)

**Comment by pwnage101** [comment]
o, just saw that this is on the wrong API version.

File: enterprise_subsidy/apps/api/v1/views/transaction.py [hunk 2]
```diff
@@ -65,6 +65,10 @@ class TransactionViewSet(
     lookup_field = "uuid"
     serializer_class = TransactionSerializer
     pagination_class = TransactionListPaginator
+    filter_backends = [filters.SearchFilter]
```
Comment (adamstankiewicz) [inline enterprise_subsidy/apps/api/v1/views/transaction.py:68]: [suggestion, non-blocking] There will be ordering on the `Spent` table backed by this `transactions` API. While we're in here, we might also consider adding the `filters.OrderingFilter` (unless that's slated for another ticket, ofc!)?

image

**Review by adamstankiewicz — APPROVED** [review]
LGTM (assuming CI is passing and these changes will soon also apply to the v2 endpoint from the Slack thread). One additional, semi-related question, non-blocking.

File: enterprise_subsidy/apps/api/v1/views/transaction.py (hunk 2 above)
Comment (johnnagro) [reply enterprise_subsidy/apps/api/v1/views/transaction.py:68]: thats a great call out - not sure if there is an existing ticket. i'll talk to the PM

**Pull request merged**